### DISCUSSION TITLE
Display destination or | and waypoint info in navigation

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#228] [Q&A] [answered]
Now we display the distance / time to the destination **or** waypoint in the right navigation panels.
Tap to switch between destination or waypoint and long press to select the displayed waypoint.

We can have another option for those who want to see the distance / time to destination **and** waypoint at the same time.
Tap the time panels to switch (separately) between remaining or arrival time, long press to select the displayed waypoint.

If you want we can change the order of the 4 panels.

<img src="https://github.com/devemux86/cruiser/assets/3484020/4285604b-7e74-4c94-8979-2dd6f1a19b3a" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/5d7e08d6-a506-4249-92d0-aa0fbebd839f" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/8cf63597-aace-4f29-97b0-38ef15af1ae9" width="33%"/>

**Comment by MotoUKRider** [comment]
This is nice - I hope the top-bar Waypoint info is remaining?

**Reply by devemux86** [reply]
> the top-bar Waypoint info is remaining?

It remains, as it can also show the waypoint name.
It can occupy vertical space, so I thought to offer more options in the side panels.

Do we need detailed options, like display only the 2 distance panels or all 4 distance + time panels?

**Reply by MotoUKRider** [reply]
That's good - I like to see the distance+name of the next Waypoint in the top-bar, and it shows well in landcape mode :)

The number of available panels may have to be governed by the space available (less space in landscape, of course). If the options are there to turn panels on/off, then users can decide how crowded they want the screen.

Throwing a suggesting into the mix, perhaps a popup dialog (like the Waypoint "start-from") that shows the Route/Waypoints distance/time plus average speed, distance travelled etc in a larger form - ie: for quick recall during a stop like a layby or traffic lights?

**Reply by devemux86** [reply]
@MotoUKRider

Do you mean a statistics screen for the already traveled route?
Or something for the untraveled route ahead of us?

**Reply by MotoUKRider** [reply]
Both, I guess! Perhaps a 3x3 (or whatever panels you can display at the time) with panels offering that information.
I *sometimes* have to stare longer at a panel icon to recall what the value means - maybe a text title above it on a larger "summary" window would help?

**Reply by devemux86** [reply]
Statistics for the traveled route / recorded track is already in my plans. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
Display options would be good because in landscape mode there could be a lack of space depending of the font size.

**Comment by devemux86** [comment]
Final preview of the new detailed display and voice settings for navigation:

<img src="https://github.com/devemux86/cruiser/assets/3484020/0a29ea10-d507-4a5f-9313-9e15dd8f21c7" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/351a76cc-0088-47ba-b25c-37cd8da454d2" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/7e6255dd-2903-4430-b632-2cb592c2508e" width="33%"/>

**Reply by Ciutnik** [reply]
I'm thinking, if you choose BRouter as the routing engine, it would be good if the options related to street names were deactivated (e.g. greyed out) automatically. This may be important in the future when several routers can be used on one route. This will prevent the display of dashes instead of street names for a section calculated by BRouter.

**Reply by devemux86** [reply]
> if you choose BRouter as the routing engine, it would be good if the options related to street names were deactivated (e.g. greyed out) automatically.

@Ciutnik nice suggestion, it will help users avoid confusion. I will see it.

**Reply by 0709wiwiwi** [reply]
> it would be good if the options related to street names were deactivated (e.g. greyed out) automatically.

Still, keep this in mind. 
What if you prefer to display the Via point name together with a possible turn?
  
Example Via trkpt content in a gpx navigation track. A gpx trkpt with name-cmt-sym-type.
![Example trkpt  combi sym turn with type via](https://github.com/devemux86/cruiser/assets/54167201/77b708b9-bff1-4eaa-b6cf-2b5110ed6cb1)
In Cruiser, the panel is used for the Via point name text display together with a (u)-turn instruction.
![U-turn plus Via Point display](https://github.com/devemux86/cruiser/assets/54167201/22ec0ba4-d9c3-4b73-866d-78fb36abc571)
Cruiser TTS  than uses the name plus the turn instruction.
@ Ciutnik.
_**I thought you were also familiar with Locus map ?**_ 
Locus TTS for a Via point is by the name.
Locus uses a similar top panel is for the cmt display. 
![Example trkpt combi sym any and type via](https://github.com/devemux86/cruiser/assets/54167201/d1c1b8b3-6cbc-4f92-8890-fdd858e49fcd)
If the cmt is empty than the panel is filled by the name text.
![Via Point Danger area Locus](https://github.com/devemux86/cruiser/assets/54167201/87d70899-d947-4275-a26f-6d703c5b9879)

Cruiser <=> Locus
_I do also refer to Locus map for the next reasons.
Efforts are made to achieve the best possible mutual compatibility with the gpx navigation track file exchanges.
Which in the GPS world is rather an oddity and not self-evident as this aspect is almost always neglected._

Willy.

**Reply by devemux86** [reply]
@0709wiwiwi this is to fill the empty panel: via points do not have streets, so it shows their name instead.

But via points have their panel if we want to show their name (and distance).
And next turn is always displayed anyway in the large panel at top left.

@Ciutnik the simplest solution is to check if the route (calculated or imported) has street names.
Then show or hide the street panels to avoid wasting the space on the screen with empty panels.

**Reply by Ciutnik** [reply]
> @0709wiwiwi 
I thought you were also familiar with Locus map ?

Yes, Locus (also Androzic) was my favorite map app on Android when it was OziExplorer on PC. Does anyone else remember good old Ozi?... 😢 But I admit that I never used Locus as navigator seriously. Just for testing, out of curiosity.

> @devemux86 
show or hide the street panels to avoid wasting the space on the screen with empty panels.

Currently, I only use BRouter, so I have the street names permanently hidden. But recently I was using interchangeably with GraphHopper and had to remember to turn them on/off. 
Hence the idea that street names should be automatically turned off when BRouter is used, when it is turned on for other routers. Just so as not to waste space with empty panel. Especially when it will be possible to use different routing engines for different sections of one route.

**Reply by 0709wiwiwi** [reply]
> @Ciutnik the simplest solution is to check if the route (calculated or imported) has street names.

👍 This would solve the issue 100% automatically and independently of the router engine source.  
As by a gpx navtrack this panel is fed and used for the streetname provided by the trkpt cmt element.  
Very unusually  I know the trkpt cmt element can also be filled with a text comment that is exceptionally not the street name. 
Well it can so be displayed at least In the Via points in Locus.  Roger ?  I suppose by the check as offered the issue is solved fair enough.

**Reply by devemux86** [reply]
> the simplest solution is to check if the route (calculated or imported) has street names.

Implemented in [Cruiser 3.0.26](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by devemux86** [comment, marked as answer]
Display of destination OR / AND waypoint info in navigation is available in [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Available in "Settings | Navigation | Display | Distance / Time".
